src/node/internal/zlib.d.ts

```diff
@@ -2,6 +2,24 @@ import { owner_symbol, type Zlib } from 'node-internal:internal_zlib_base';
 
 export function crc32(data: ArrayBufferView, value: number): number;
 
+type CompressCallback = (
+  error: string | null,
+  result: ArrayBuffer | null
+) => void;
+
+export function crc32(data: ArrayBufferView | string, value: number): number;
+export function zlibSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions,
+  mode: number
+): ArrayBuffer;
+export function zlibWithCb(
+  data: ArrayBufferView | string,
+  options: ZlibOptions,
+  mode: number,
+  cb: CompressCallback
+): ArrayBuffer;
```

Comment:
Is this separate `zlibWithCb` actually needed? Couldn't the `internal_zlib.ts` just call `zlibSync` directly?

Reply:
@anonrig suggested doing it this way for performance reasons

Reply:
I think you misunderstood me here. We can have a single function on C++ land, and trigger the callback if the 4th argument is callback. If not, it can just return a value and act like sync.

Reply:
I'm not sure I see the advantage of having a single function on the C++ side. It would have a more confusing return type (Some array if no callback, None if there is a callback). There's very little code to reuse either.